Add hardware requirements for package compilation (#1093)
JuliaLang/PackageCompiler.jl: docs/src/index.md

A suitable compiler will be automatically installed the first time it is needed.

## Hardware requirements
If you include many packages in your image, a large amount of RAM is needed. For single threaded operation 16 GB are common,
for multithreaded compilation often 32 GB are required. But this is RAM + swap file size, so if you have only 8 GB RAM using
a 8 GB swap file will make it work.

> If you are low of RAM, ensure to use only one thread for building the package. This can be achieved with the command:

Reviewer: ```suggestion
If you are low on RAM, use only one thread to build the package. This can be achieved with the command:
```
